warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/handler/shard/ShardedDataTypeHandler.java

```diff
@@ -679,19 +680,37 @@ protected Value createBloomFilter(RawRecordContainer event, Multimap<String,Norm
         
         if (null != maskedFieldHelper && maskedFieldHelper.contains(column)) {
             // These Keys are for the index, so if they are masked, we really want to use the normalized masked values
+            // It was observed that the normalized mask values aren't coming back reversed, so account for that before creating the row.
             final String normalizedMaskedValue = helper.getNormalizedMaskedValue(column);
             
             Text colf = new Text(column);
             Text colq = new Text(shardId);
             TextUtil.textAppend(colq, event.getDataType().outputName(), helper.getReplaceMalformedUTF8());
             
-            // Dont create index entries for empty values
-            if (!StringUtils.isEmpty(normalizedMaskedValue)) {
-                // Create a key for the masked field value with the masked visibility
-                Key k = this.createIndexKey(normalizedMaskedValue.getBytes(), colf, colq, maskedVisibility, event.getDate(), false);
+            // if this method was called with the intention to create reverse index keys, ensure the masked values are reversed.
+            if (isReverse) {
```

Comment:
I think this whole section could be much simpler by switching the if statements:
```
if (!StringUtils.isEmpty(normalizedMaskedValue)) {
   if (isReverse) {
     normalizedMaskedValue = new StringBuilder().append(normalizedMaskedValue).reverse().toString();
   }
   Key k = this.createIndexKey(normalizedMaskedValue.getBytes(), colf, colq, maskedVisibility, event.getDate(0, false);
   BulkIngestKey bKey = BulkIngestKey(tableName, k);
   values.put(bkey, indexValue);
}
```

Reply:
I'll adjust the ordering on the next commit.